pallets/claims/src/migration.rs

```diff
@@ -0,0 +1,61 @@
+use super::*;
+use hex::FromHex;
+use primitives::Balance;
+
+pub fn import_initial_claims<T: Config>(claims_data: &[(&'static str, Balance)]) -> frame_support::weights::Weight {
+	if PalletVersion::get() == StorageVersion::V1EmptyBalances {
+		frame_support::debug::info!(" >>> Adding claims to the storage");
+		for (addr, amount) in claims_data.iter() {
+			let balance: BalanceOf<T> = T::CurrencyBalance::from(*amount).into();
+
+			Claims::<T>::insert(
+				EthereumAddress(<[u8; 20]>::from_hex(&addr[2..]).unwrap_or_else(|addr| {
+					frame_support::debug::warn!("Error encountered while migrating Ethereum address: {}", addr);
+					EthereumAddress::default().0
+				})),
+				balance,
+			);
+		}
+		PalletVersion::put(StorageVersion::V2AddClaimData);
+		T::DbWeight::get().reads_writes(2, 3)
+	} else {
+		frame_support::debug::info!(" >>> Unused migration");
+		0
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate::mock::Test;
+
+	#[test]
+	fn data_migration_should_work() {
```

Comment:
We need more tests.

- negative balances
- strings
- floats
- duplicated address
- zero balance
- overflowed balance

Also we don't handle these in migration and we probably should.

Reply:
I think it's too late to do any tests at runtime (in this case, since data migration is a one-time operation handled by responsible devs/chain maintainers).

Reply:
How do we generate the data into .rs file? @lumir-mrkva

Reply:
i agree with @Roznovjak, migration doesn't need to be tested so thoroughly, besides we can test it with real data that we'll do real migration with

Reply:
OK but then we need to definitely test the source script that generates it.

Reply:
We can create public subrepo for the script, or add the script to the `claims pallet` folder (or some `script` folder?).

Reply:
Will be resolved separately [here](https://github.com/galacticcouncil/xhdx) I suppose?